File: TEO/Mav.UI/Mav.Ui/src/app/models/admin/update-assessment.view.model.ts
import { AssessmentGradeViewModal } from "./assessment-grade.view.model";

export class UpdateAssessmentViewModel {
    public id: number;

    public yearID: number;

    public eoQualification: boolean;

    public minScore: number;

    public title: string;

    public guide: string;

    public objectives: string;

    public published: boolean;

    public roleID?: string;

    public assessmentGrade: AssessmentGradeViewModal;
}

File: TEO/Mav.Models/ViewModels/UpdateAssessmentViewModel.cs
using Mav.Data.Entities;
using System.ComponentModel.DataAnnotations;

namespace Mav.Models.ViewModels
{
    public class UpdateAssessmentViewModel
    {
        public int ID { get; set; }

        [Required]
        public int YearID { get; set; }

        public bool EoQualification { get; set; }

        [Required]
        public int MinScore { get; set; }

        [Required(ErrorMessage = "Please enter an assessment title")]
        [Display(Name = "Title")]
        [StringLength(400, ErrorMessage = "Assessment title value is too long")]
        public string Title { get; set; }

        public string Guide { get; set; }

        public string Objectives { get; set; }

        public bool Published { get; set; }

        public int? StatementID { get; set; }

        [Required]
        public string RoleID { get; set; }

        public AssessmentGradeViewModel AssessmentGrade { get; set; }
    }
}
